app/Nova/Actions/ActivateUdbOrganizer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Nova\Actions;
+
+use App\Domain\Integrations\Models\UdbOrganizerModel;
+use App\UiTPAS\Jobs\ActivateUiTPASClient;
+use App\UiTPAS\Jobs\ActivateUiTPASClientHandler;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Bus\Dispatcher;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Collection;
+use Laravel\Nova\Actions\Action;
+use Laravel\Nova\Fields\ActionFields;
+use Ramsey\Uuid\Uuid;
+
+final class ActivateUdbOrganizer extends Action
+{
+    use InteractsWithQueue;
+    use Queueable;
+
+    public $name = 'Approve UDB3 organizer request';
+
+    public function __construct(
+        private readonly Dispatcher $dispatcher,
+        private readonly ActivateUiTPASClientHandler $listener,
+    ) {
+    }
+
+    public function handle(ActionFields $fields, Collection $udbOrganizers): void
+    {
+        foreach ($udbOrganizers as $udbOrganizer) {
+            if (!$udbOrganizer instanceof UdbOrganizerModel) {
```

Comment:
This should never happen. If this were to happen, then we need to fix it.

Reply:
I agree this should never happen, and probably never will.
The reason I write code like this is twofold:
- Just pure defensive programming
- This helps the IDE understand the type of `$udbOrganizer`,  without using docblocks. 

I would prefer to keep these types of statements.

Reply:
The side effect of this defensive style is that errors go unnoticed. To help the IDE understand the type of the Collection, you can also typehint that.

Reply:
There is no error in this case breaking the code? Because I handled it with the if.
How would you typehint this without a docblock?

Reply:
I said that coding errors would go unnoticed, not that it would result in errors. If, for some reason, the provided collection does not contain elements of type `UdbOrganizerModel`, there won't be a crash. But the result would be that organizers don't get activated. You obscure the flow of the code.

Reply:
Also, this is code deep inside a domain layer we fully control. This is not something a user has entered and needs extra validation. In fully typed languages with generics this would fail when calling the method, in PHP this will fail when the method is executing.

Reply:
But fine if you want to keep it.